client/src/components/ServicesFilter/ServicesFilter.jsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import {listItemSectionBuilder, MultiSelectCheckbox, TableToolbar} from 'wix-style-react';
+import {translate} from '../../utils';
+import {isEqual} from 'lodash';
+
+export default class ServicesFilter extends React.Component {
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            selectedServices: props.filters.services
+        };
+    }
+
+    _getOptionsList = (options) => {
+        const {servicesGroups} = this.props;
+        for (const serviceGroup in servicesGroups) {
+            if (servicesGroups[serviceGroup].length) {
+                options.push(listItemSectionBuilder({title: translate(`ServicesFilter.${serviceGroup.toLowerCase()}`)}));
+                options.push(...servicesGroups[serviceGroup]);
+            }
+        }
+    };
+
+    _onServiceOptionSelect = (optionScheduleId) => {
+        this.setState({selectedServices: [...this.state.selectedServices, optionScheduleId]});
+    };
+
+    _onServiceOptionDeselect = (optionScheduleId) => {
+        this.setState({selectedServices: this.state.selectedServices.filter((selectedService) => selectedService !== optionScheduleId)});
+    };
+
+    _onServiceFilterClose = () => {
+        const {onFilterChanged, filters} = this.props;
+        if (!isEqual(filters.services.toJS().sort(), this.state.selectedServices.sort())) {
+            onFilterChanged('services', this.state.selectedServices);
+        }
+    };
+
+    render() {
+        const options = [];
```

Comment:
i don't really like this practice of pushing stuff into an array inside a render function.

just initialize the array inside the _getOptionsList function.
if you still need an optional **options** array (that already contains some options for example) you can pass still pass it as a function argument with a default value of an empty array.